src/ray/raylet/scheduling_queue.h

```diff
@@ -217,6 +222,16 @@ class SchedulingQueue {
   /// The set of currently running driver tasks. These are empty tasks that are
   /// started by a driver process on initialization.
   std::unordered_set<TaskID> driver_task_ids_;
+
+  /// \brief Return all resource demand associated with the specified task queue.
+  ///
+  /// \param task_queue The task queue for which aggregate resource demand is calculated.
+  /// \return Aggregate resource demand.
+  ResourceSet GetQueueResources(const TaskQueue &task_queue) const;
+  ResourceSet GetReadyQueueResources() const;
+  ResourceSet GetWaitingQueueResources() const;
+  ResourceSet GetRunningQueueResources() const;
```

Comment:
Are these last three methods called anywhere outside of the `SchedulingQueue`? I would probably just remove these and call their function bodies directly from `GetResourceLoad`.